traversal/procedure/ProcedureVertex.java

```diff
@@ -153,19 +176,10 @@ public boolean overlaps(Thing other, Traversal.Parameters params) {
 
         @Override
         public <ORDER extends Order> Forwardable<? extends ThingVertex, ORDER> iterator(
-                GraphManager graphMgr, Traversal.Parameters parameters, ORDER order
+                GraphManager graphMgr, Traversal.Parameters parameters, ORDER order, boolean forceValueSort
         ) {
             if (props().hasIID()) return iterateAndFilterFromIID(graphMgr, parameters, order);
-            else return iterateAndFilterFromTypes(graphMgr, parameters, order);
-        }
-
-        <ORDER extends Order> Forwardable<? extends ThingVertex, ORDER> filter(
-                Forwardable<? extends ThingVertex, ORDER> iterator, Traversal.Parameters params
-        ) {
-            iterator = filterTypes(iterator);
-            if (props().hasIID()) iterator = filterIID(iterator, params);
-            if (!props().predicates().isEmpty()) iterator = filterPredicates(filterAttributes(iterator), params);
-            return iterator;
+            else return iterateAndFilterFromTypes(graphMgr, parameters, order, forceValueSort);
         }
 
         <ORDER extends Order> Forwardable<? extends ThingVertex, ORDER> iterateAndFilterFromIID(
```

Comment:
the following methods are all the `filter + iterate/merge` methods used by `ProcedureEdge`. Additionally we use them when trying to create an iterator for the vertex, without the edge.